test/unit/test_data_serialization.py

```diff
@@ -260,68 +267,260 @@ def test_circuit_metadata(self):
+        self.assertEqual(pub1.shots, pub2.shots)
+
+    def assert_pub_results_equal(self, pub_result1, pub_result2):
+        """Tests that two PubResult objects are equal"""
+        self.assert_data_bins_equal(pub_result1.data, pub_result2.data)
+        self.assertEqual(pub_result1.metadata, pub_result2.metadata)
+
+    def assert_primitive_results_equal(self, primitive_result1, primitive_result2):
+        """Tests that two PrimitiveResult objects are equal"""
+        self.assertEqual(len(primitive_result1), len(primitive_result2))
+        for pub_result1, pub_result2 in zip(primitive_result1, primitive_result2):
+            self.assert_pub_results_equal(pub_result1, pub_result2)
+        self.assertEqual(primitive_result1.metadata, primitive_result2.metadata)
+
+    # Data generation methods
+
+    @staticmethod
+    def make_test_data_bins():
+        """Generates test data for DataBin test"""
+        result_bins = []
+        data_bin_cls = make_data_bin([("alpha", np.ndarray), ("beta", np.ndarray)], shape=(10, 20))
+        alpha = np.empty((10, 20), dtype=np.uint16)
+        beta = np.empty((10, 20), dtype=int)
+        my_bin = data_bin_cls(alpha, beta)
+        result_bins.append(my_bin)
+        return result_bins
+
+    @staticmethod
+    def make_test_estimator_pubs():
+        """Generates test data for EstimatorPub test"""
+        pubs = []
+        params = (Parameter("a"), Parameter("b"))
+        circuit = QuantumCircuit(2)
+        circuit.rx(params[0], 0)
+        circuit.ry(params[1], 1)
+        parameter_values = BindingsArray(data={params: np.ones((10, 2))})
+        observables = ObservablesArray([{"XX": 0.1}])
+        precision = 0.05
+
+        pub = EstimatorPub(
+            circuit=circuit,
+            observables=observables,
+            parameter_values=parameter_values,
+            precision=precision,
+        )
+        pubs.append(pub)
+        return pubs
+
+    @staticmethod
+    def make_test_sampler_pubs():
+        """Generates test data for SamplerPub test"""
+        pubs = []
+        params = (Parameter("a"), Parameter("b"))
+        circuit = QuantumCircuit(2)
+        circuit.rx(params[0], 0)
+        circuit.ry(params[1], 1)
+        circuit.measure_all()
+        parameter_values = BindingsArray(data={params: np.ones((10, 2))})
+        shots = 1000
+
+        pub = SamplerPub(
+            circuit=circuit,
+            parameter_values=parameter_values,
+            shots=shots,
+        )
+        pubs.append(pub)
+        return pubs
+
+    @staticmethod
+    def make_test_pub_results():
+        """Generates test data for PubResult test"""
+        pub_results = []
+        data_bin = make_data_bin((("a", float), ("b", int)))
+        pub_result = PubResult(data_bin(a=1.0, b=2))
+        pub_results.append(pub_result)
+        pub_result = PubResult(data_bin(a=1.0, b=2), {"x": 1})
+        pub_results.append(pub_result)
+        return pub_results
+
+    @staticmethod
```

Comment:
This test suite won't run because these are static methods and they can't be passed into `@data`

`python -m unittest test/unit/test_data_serialization.py`